lib/pure/json.nim

```diff
@@ -964,16 +964,28 @@ proc escapeJson*(s: string): string =
   result.add("\"")
 
 proc toPretty(result: var string, node: JsonNode, indent = 2, ml = true,
-              lstArr = false, currIndent = 0) =
+              lstArr = false, currIndent = 0, sort_keys = false) =
   case node.kind
   of JObject:
     if currIndent != 0 and not lstArr: result.nl(ml)
     result.indent(currIndent) # Indentation
     if node.fields.len > 0:
       result.add("{")
       result.nl(ml) # New line
+      var get_pairs = (block:
+        iterator it: (string, JsonNode) {.closure.} =
+          if sort_keys:
+            for key, val in sorted_pairs(node.fields, system.cmp[string]):
```

Comment:
'cmp' without the string specializiation produces better code. (And I hope it compiles)